pybossa/api/__init__.py

```diff
@@ -417,6 +420,92 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name=None):
+    """API endpoint for loading account user preferences.
+    Returns a JSON object containing the user account preferences.
+    """
+    if not user_name:
+        return abort(404)
+
+    user = user_repo.get_by_name(user_name)
+    if not user:
+        return abort(404)
+
+    try:
+        (can_update, disabled_fields, hidden_fields) = can_update_user_info(current_user, user)
+    except Exception:
+        return abort(404)
+
+    if not can_update:
+        return abort(403)
+
+    user_preferences = get_user_pref_metadata(user_name)
+
+    if not user_preferences and user_preferences != {}:
+        return abort(500)
+
+    return Response(json.dumps(user_preferences), mimetype="application/json")
+
+
+@jsonpify
+@login_required
+@csrf.exempt
+@blueprint.route('/preferences/<user_name>', methods=['POST'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def update_user_preferences(user_name=None):
+    """API endpoint for updating account user preferences.
+    Returns a JSON object containing the updated user account preferences.
+    """
+    if not user_name:
+        return abort(404)
+
+    user = user_repo.get_by_name(user_name)
+    if not user:
+        return abort(404)
+
+    try:
+        (can_update, disabled_fields, hidden_fields) = can_update_user_info(current_user, user)
```

Comment:
it seems disabled and hidden fields to be handled for not to be updated as required.

Reply:
Right, those two parameters are not being used, but were included for consistency with the existing methods in the code.

Reply:
Looks like disabled field can be populated via `can_update_user_info ` depending on user access and need to be considered here. Drop disabled field from update as user doesn't has sufficient privilege to update it.

Reply:
Fixed via https://github.com/bloomberg/pybossa/pull/675/commits/f0eb0e90d8158450cca12dba8a3208cc335403b6